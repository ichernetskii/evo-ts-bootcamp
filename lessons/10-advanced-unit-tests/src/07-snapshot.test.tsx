import React from "react"
import { render } from "@testing-library/react"
import { Message } from "./07-snapshot"

describe("07-snapshot", () => {
  describe("Message", () => {
    it("has a proper content #1", () => {
      const { container } = render(
        <Message
          message={"hello"}
          color={"#f00"}
        />
      )
      expect(container.firstChild).toMatchSnapshot()
    })

    it("has a proper content #2", () => {
      const { container } = render(
        <Message
          message={"hello"}
        />
      )
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
