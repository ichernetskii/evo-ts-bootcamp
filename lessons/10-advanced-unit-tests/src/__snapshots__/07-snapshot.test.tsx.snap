// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`07-snapshot Message has a proper content #1 1`] = `
<div
  style="color: rgb(255, 0, 0);"
>
  hello
</div>
`;

exports[`07-snapshot Message has a proper content #2 1`] = `
<div>
  hello
</div>
`;
